packages/web/components/swap-tool/trade-details.tsx

```diff
@@ -91,42 +82,15 @@ export const TradeDetails = observer(
       });
     }, [makerFee]);
 
-    const gasFeeError = useMemo(() => {
-      if (!!gasAmount && !gasError) return;
-
-      return isOneClickTradingEnabled
-        ? t("swap.gas.oneClickTradingError")
-        : t("swap.gas.error");
-    }, [gasAmount, isOneClickTradingEnabled, gasError, t]);
-
-    const GasEstimation = useMemo(() => {
-      return !!gasFeeError ? (
-        <GenericDisclaimer
-          title={t("swap.gas.gasEstimationError")}
-          body={gasFeeError}
-        >
-          <span className="flex items-center gap-1">
-            <Icon
-              id="question"
-              width={24}
-              height={24}
-              className="scale-75 text-osmoverse-300"
-            />{" "}
-            {t("swap.gas.unknown")}
-          </span>
-        </GenericDisclaimer>
-      ) : (
-        <span
-          className={classNames(
-            "inline-flex items-center gap-1 text-osmoverse-100",
-            { "animate-pulse": isGasLoading }
-          )}
-        >
-          <Icon id="gas" width={16} height={16} />
-          {gasAmount && gasAmount.toString()}
-        </span>
-      );
-    }, [gasAmount, isGasLoading, gasFeeError, t]);
+    useEffect(() => {
```

Comment:
We can avoid that useEffect, instead we can do the following things:

1) const [outAsBase, setOutAsBase] = useState(tab === "buy");
2) When we use `TradeDetails` component inside `PlaceLimitTool`, you can pass a prop to `TradeDetails` that is `key={tab}`, you already have the tab value on the parent

So we avoid side effects, and just re-render the component if the tab change, it's a suggested behaviour from react docs (for example for form reset)